### PR TITLE
fix tree data algorithms exception

### PR DESCRIPTION
**Pull request by downdawn**

### CODE REVIEW

**Comment by downdawn** [comment]
please review

File: backend/sql/init_test_data.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 INSERT INTO fba.sys_dept (id, name, level, sort, leader, phone, email, status, del_flag, parent_id, created_time, updated_time)
 VALUES (1, 'test', 0, 0, null, null, null, 1, 0, null, '2023-06-26 17:13:45', null);
 
-insert into fba.sys_menu (id, name, level, sort, icon, path, menu_type, component, perms, status, remark, parent_id, created_time, updated_time, 'show', cache)
+insert into fba.sys_menu (id, name, level, sort, icon, path, menu_type, component, perms, status, remark, parent_id, created_time, updated_time, `show`, cache)
```
Comment (wu-clan) [inline backend/sql/init_test_data.sql:4]: This is the latest test data, can you update it together?
```sql
insert into fba.sys_menu (id, name, level, sort, icon, path, menu_type, component, perms, status, remark, parent_id, created_time, updated_time, `show`, cache, title)
values  (1, 'test', 0, 0, null, null, 0, null, null, 1, null, null, '2023-06-26 17:13:45', null, 0, 1, '测试'),
        (2, 'dashboard', 0, 0, 'icon-dashboard', '/dashboard', 0, '/dashboard/workplace/index.vue', null, 1, null, null, '2023-06-30 10:10:34', null, 1, 1, '仪表盘'),
        (3, 'Workplace', 0, 0, null, '/workplace', 1, null, null, 1, null, 2, '2023-06-30 10:11:40', null, 1, 1, '工作台'),
        (4, 'arcoWebsite', 0, 0, 'icon-link', 'https://arco.design', 0, null, null, 1, null, null, '2023-06-30 10:13:04', null, 1, 1, 'arco官网'),
        (5, 'log', 0, 0, 'icon-bug', '/log', 0, null, null, 1, null, null, '2023-06-30 10:13:54', null, 1, 1, '日志'),
        (6, 'Login', 0, 0, null, '/login', 1, '/log/login/index.vue', null, 1, null, 5, '2023-06-30 10:14:23', null, 1, 1, '登录日志'),
        (7, 'faq', 0, 0, 'icon-question-circle', 'https://arco.design/vue/docs/pro/faq', 0, null, null, 1, null, null, '2023-06-30 10:14:56', null, 1, 1, '常见问题'),
        (8, 'system', 0, 0, 'icon-settings', '/system', 0, '', null, 1, null, null, '2023-07-04 10:52:48', '2023-07-04 04:56:00', 1, 1, '系统管理'),
        (9, 'SysMenu', 0, 0, '', '/sysmenu', 1, '/system/menu/index.vue', null, 1, null, 8, '2023-07-04 10:55:02', '2023-07-04 04:56:25', 1, 1, '系统菜单');
```

**Review by wu-clan — APPROVED** [review]
LGTM.

File: backend/sql/init_test_data.sql (hunk 1 above)
Comment (downdawn) [reply backend/sql/init_test_data.sql:4]: I think it is better to update it separately and synchronize with the front end.
arcoWebsite and faq are only temporary, the sorting can be set larger

Comment (wu-clan) [reply backend/sql/init_test_data.sql:4]: OK

**Pull request merged**